Dependencies/UrlLib/Include/UrlLib/UrlLib.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <memory>
+#include <arcana/threading/task.h>
+
+namespace UrlLib
+{
+    enum class UrlStatusCode
+    {
+        None = 0,
+        Ok = 200,
+    };
+
+    enum class UrlMethod
+    {
+        Get,
+    };
+
+    enum class UrlResponseType
+    {
+        String,
+        Buffer,
+    };
+
+    class UrlRequest final
+    {
+    public:
+        UrlRequest();
+
+        UrlRequest(const UrlRequest&);
+
+        UrlRequest(UrlRequest&&);
+
+        ~UrlRequest();
+
+        void Abort();
+
+        void Open(UrlMethod method, std::string url);
+
+        UrlResponseType ResponseType() const;
+
+        void ResponseType(UrlResponseType value);
+
+        arcana::task<void, std::exception_ptr> SendAsync();
+
+        UrlStatusCode StatusCode() const;
+
+        gsl::cstring_span<> ResponseUrl() const;
```

Comment:
Are you just getting gsl through arcana?

Reply:
Yes.